src/helpers/query-daily-data.tsx

```diff
@@ -0,0 +1,94 @@
+﻿import { add } from "date-fns";
+import getDayKey from "./get-day-key";
+import { DailyDataType, DailyDataTypeDefinition } from "./daily-data-types";
+import allTypeDefinitions from "./daily-data-types/all";
+import { FontAwesomeSvgIcon } from "react-fontawesome-svg-icon";
+import { faPersonRunning } from "@fortawesome/free-solid-svg-icons";
+import React from "react";
+import { defaultFormatter } from "./daily-data-types/formatters";
+
+export type DailyDataQueryResult = { [key: string]: number };
+export type DailyDataProvider = (startDate: Date, endDate: Date) => Promise<DailyDataQueryResult>;
+export type DailyDataAvailabilityCheck = (modifiedAfter?: Date) => Promise<boolean>;
+
+let dailyDataTypes = new Map(
+	allTypeDefinitions.map((typeDefinition) => [typeDefinition.type, typeDefinition])
+);;
+
+export function registerDailyDataProvider(type: string, provider: DailyDataProvider, availabilityCheck: DailyDataAvailabilityCheck) {
+	dailyDataTypes.set(type, {
+		type: type,
+		getLabel: () => type,
+		dataProvider: provider,
+		availabilityCheck: availabilityCheck,
+		icon: <FontAwesomeSvgIcon icon={faPersonRunning} />,
+		formatter: defaultFormatter,
+		previewDataRange: [0, 10000]
```

Comment:
Same question here.  Should the icon and this data range be optional params with these defaults?